dvc/utils/fs.py

```diff
@@ -31,3 +32,24 @@ def get_mtime_and_size(path):
     # State of files handled by dvc is stored in db as TEXT.
     # We cast results to string for later comparisons with stored values.
     return str(int(nanotime.timestamp(mtime))), str(size)
+
+
+class BasePathNotInCheckedPathException(DvcException):
+    def __init__(self, path, base_path):
+        msg = "Path: {} does not overlap with base path: {}".format(
+            path, base_path
+        )
+        super(DvcException, self).__init__(msg)
+
+
+def contains_symlink_up_to(path, base_path=""):
+    if base_path not in path:
+        raise BasePathNotInCheckedPathException(path, base_path)
+
+    if System.is_symlink(path):
```

Comment:
What if repo itself is a symlink? Seems like it should be `False`, right? Could you add a test for this case, please?